core/drivers/crypto/hisilicon/hisi_qm.c

```diff
@@ -428,11 +437,67 @@ static enum hisi_drv_status qm_alloc(struct hisi_qm *qm)
 	memset(qm->cqc, 0, cqc_size);
 	qm->cqc_dma = virt_to_phys(qm->cqc);
 
-	qm->qp_array = calloc(qm->qp_num, sizeof(struct hisi_qp));
+	return HISI_QM_DRVCRYPT_NO_ERR;
+
+	free(qm->cqc);
+free_sqc:
+	free(qm->sqc);
+	return ret;
+}
+
+static enum hisi_drv_status hisi_qm_alloc_rsv_xqc(struct hisi_qm *qm)
```

Comment:
This function is copy & paste of `hisi_qm_alloc_xqc()`, please find a way to reuse `hisi_qm_alloc_xqc()` instead.

Reply:
solved